test/parallel/test-http-server-connections-checking-leak.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+// Flags: --expose-gc
+
+// Check that creating a server without listening does not leak resources.
+
+const common = require('../common');
+const onGC = require('../common/ongc');
+const Countdown = require('../common/countdown');
+
+const http = require('http');
+const max = 100;
+
+// Note that Countdown internally calls common.mustCall, that's why it's not done here.
+const countdown = new Countdown(max, () => {});
+
+for (let i = 0; i < max; i++) {
+  const server = http.createServer((req, res) => {});
+  onGC(server, { ongc: countdown.dec.bind(countdown) });
+}
+
+setTimeout(() => {
```

Comment:
Any reason for this?

Reply:
Yes: on my machine `global.gc` right after the for loop was not collecting the last server yet.

Reply:
Is `setImmediate()` sufficient? Waiting 500 ms seems a bit too much.

Reply:
I'm not sure about it. Probably it is, but I was being conservative against flakiness.

Reply:
It works in all machines I've tested it on (Windows, Linux, macOS). We can always increase it later.

Reply:
Anyway this was the reason for the `setInterval()` suggestion above. To make the test complete as soon as possible without waiting too much or too less.

Reply:
I see. Let me swap with setImmediate.